pennylane_lightning/core/_measurements_base.py

```diff
@@ -203,7 +203,7 @@ def get_measurement_function(
             Callable: function that returns the measurement result
         """
         if isinstance(measurementprocess, StateMeasurement):
-            if isinstance(measurementprocess, ExpectationMP):
+            if isinstance(measurementprocess, ExpectationMP) and measurementprocess.obs is not None:
```

Comment:
AFAIK `op` is not optional in `expval(op)`. Why to bother and check this here?

Reply:
Thanks @mudit2812! Is there any unit test for this cause I can revert this and still get all checks in `test_eval_jaxpr` pass?